pkg/cmd/plugin_install.go

```diff
@@ -68,10 +88,39 @@ func (o *pluginInstallOptions) complete(args []string) error {
 	return nil
 }
 
+func (o *pluginInstallOptions) newInstallerForSource() (installer.Installer, error) {
+	// Check if source is an OCI registry reference
+	if strings.HasPrefix(o.source, fmt.Sprintf("%s://", registry.OCIScheme)) {
+		// Build getter options for OCI
+		var options []getter.Option
+
+		if o.certFile != "" || o.keyFile != "" || o.caFile != "" {
+			options = append(options, getter.WithTLSClientConfig(o.certFile, o.keyFile, o.caFile))
+		}
+
+		if o.insecureSkipTLSverify {
+			options = append(options, getter.WithInsecureSkipVerifyTLS(o.insecureSkipTLSverify))
+		}
+
+		if o.plainHTTP {
+			options = append(options, getter.WithPlainHTTP(o.plainHTTP))
+		}
+
+		if o.username != "" || o.password != "" {
+			options = append(options, getter.WithBasicAuth(o.username, o.password))
```

Comment:
Not that I have a hard opinion on this, but I'm wondering if these `getter.WithBasicAuth()` or other `With` function calls could contain the logic for `if o.username != "" || if o.password != ""`

Then this `newInstallerForSource` function would just be a registration of options, like

```
options = append(options, getter.WithTLSClientConfig(o.certFile, o.keyFile, o.caFile))
options = append(options, getter.WithInsecureSkipVerifyTLS(o.insecureSkipTLSverify))
options = append(options, getter.WithPlainHTTP(o.plainHTTP))
options = append(options, getter.WithBasicAuth(o.username, o.password))

```